ukpn/scripts/resample_data.py

```diff
@@ -0,0 +1,94 @@
+"""Function to resample the irreggular time series data into regular"""
+import random
+from typing import Optional, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from dateutil.parser import parse
+
+
+def resample_dataframe(
+    path_to_file: str, intervals: Optional[str] = "5Min", resample: Optional[bool] = False
+) -> pd.DataFrame:
+    """This function resamples a time series into regular intervals
+
+    Args:
+        path_to_file: Enter the absolute path to the csv file
+        intervals: Enter the intervals that you want this data to have
+        resample: Option to resample the dataframe
+
+    """
+    # Reading the csv data from the path
+    df = pd.read_csv(path_to_file, header=None, names=["date_time", "bad_data"], sep=",")
+
+    # # Convert data values from str into int
+    df["bad_data"] = pd.to_numeric(df["bad_data"], errors="coerce")
+
+    # Interpolate with padding
+    df["bad_data"] = df["bad_data"].interpolate(method="pad", limit=2)
```

Comment:
```suggestion
```

I'd recommend removing this line if we are resampling below. If we aren't resampling, then we want the raw data to be the raw data, so just remove the bad data points and leave the rest as is.

Reply:
I think, this step just fills in values where there is a `Bad Data` entry like the one below, using padding method, I don't think this is a resampling.

fake_data
25-Nov-17 09:59:57,-113.678
25-Nov-17 10:27:34,Bad Data
25-Nov-17 11:36:48,-11.5678